test/runner.js

```diff
@@ -208,6 +208,29 @@ describe('Runner', function () {
         });
     });
 
+    it('debug domain error', function (done) {
+
+        var script = Lab.script();
+        script.experiment('test', function () {
+
+            script.test('a', function (finished) {
+
+                setImmediate(function () {
+
+                    throw new Error('throwing stack later');
+                });
+
+                finished();
+            });
+        });
+
+        Lab.execute(script, { debug: true }, null, function (err, notebook) {
```

Comment:
There are some console artifacts from the `console.log` that was added to the `onError` function. Should I be pushing a new error to `state.report.errors` instead?